events.go

```diff
@@ -175,6 +176,73 @@ func (defaultEventHandlers) RegisterChatMessage(ec *EventCollector) {
 	})
 }
 
+func (defaultEventHandlers) RegisterGrenadeEvents(ec *EventCollector) {
+  ec.AddHandler(func(e events.SmokeStart) {
+    eb := buildEvent(rep.EventSmokeStart)
+    eb.floatAttr("x", e.Position.X)
+    eb.floatAttr("y", e.Position.Y)
+    eb.floatAttr("z", e.Position.Z)
+    eb.intAttr(rep.AttrKindEntityID, e.Thrower.EntityID);
```

Comment:
I think we will want a new `AttrKind`

probably `AttrKindThrowerID`

because there are technically also grenade entities. When I used the word `entity` for players in other parts was probably a mistake by me, but that's a topic for another day.

Reply:
So, how should the resulting JSON look like? We currently have something like this.

```
{"name":"smoke_start","attrs":[{"key":"x","numVal":-291.2807312011719},{"key":"y","numVal":1448.5047607421875},{"key":"z","numVal":-26.500001907348633},{"key":"entityId","numVal":14}]}
```

I guess the goal  should be something like this:
```json
{"name":"smoke_start","attrs":[{"key":"x","numVal":-291.2807312011719},{"key":"y","numVal":1448.5047607421875},{"key":"z","numVal":-26.500001907348633},{"key":"throwerEntityId","numVal":14}]}
```
(Renamed EntityID to throwerId)
Or more like this?
```json
{
"name":"smoke_start",
"attrs":[
  { 
   "key":"position",
   "objVal": {"x": -291.2807312011719, "y": 1448.5047607421875, "z":26.500001907348633 }]}
```
(with an object value type)

Reply:
I might have gone with `throwerId` for the sake of shortness but `throwerEntityId` is good as well 👍